aiida_common_workflows/workflows/dissociation.py

```diff
@@ -0,0 +1,216 @@
+
+
+def validate_relax(value, _):
+    """Validate the `generator_inputs.relax_type` input."""
+    if value is not RelaxType.NONE:
+        return '`generator_inputs.relax_type`. Only `RelaxType.NONE` supported.'
+
+
+@calcfunction
+def set_distance(molecule: orm.StructureData, distance: orm.Float) -> orm.StructureData:
+    """
+    Move the second site of the molecule to meet a target distance
+    between the two sites of the molecule.
+    """
+    import numpy as np
+    vector_diff = np.array(molecule.sites[1].position) - np.array(molecule.sites[0].position)
+    versor_diff = vector_diff / np.linalg.norm(vector_diff)
+    molecule_new = molecule.clone()
+    new_position = np.array(molecule.sites[0].position) + distance.value * versor_diff
+    molecule_new.attributes['sites'][1]['position'] = new_position
+    return molecule_new
+
+
+class DissociationCurveWorkChain(WorkChain):
+    """Workflow to compute the dissociation curve of for a given diatomic molecule."""
+
+    @classmethod
+    def define(cls, spec):
+        # yapf: disable
+        super().define(spec)
+        spec.input('molecule', valid_type=orm.StructureData, validator=validate_molecule, help='The input molecule.')
+        spec.input('distances', valid_type=orm.List, required=False, validator=validate_distances,
+            help='The list of dinstances in Ang at which the total energy of the molecule should be computed.')
+        spec.input('distances_count', valid_type=orm.Int, default=lambda: orm.Int(20),
+            validator=validate_distances_count,
+            help='The number of points to compute in the dissociation curve.')
+        spec.input('distance_min', valid_type=orm.Float, default=lambda: orm.Float(0.5),
+            validator=validate_distance_min,
+            help='The minimum tested dinstance in Ang.')
+        spec.input('distance_max', valid_type=orm.Float, default=lambda: orm.Float(3),
+            validator=validate_distance_max,
+            help='The maximum tested distance in Ang.')
+        spec.input_namespace('generator_inputs', dynamic=True,
+            help='The inputs that will be passed to the inputs generator of the specified `sub_process`.')
+        spec.input('generator_inputs.calc_engines', valid_type=dict, non_db=True)
+        spec.input('generator_inputs.protocol', valid_type=str, non_db=True,
+            help='The protocol to use when determining the workchain inputs.')
+        spec.input('generator_inputs.relax_type', valid_type=RelaxType, non_db=True, validator=validate_relax,
+            help='The type of relaxation to perform.')
+        spec.input('generator_inputs.spin_type', valid_type=SpinType, required=False, non_db=True,
+            help='The type of spin for the calculation.')
+        spec.input('generator_inputs.electronic_type', valid_type=ElectronicType, required=False, non_db=True,
+            help='The type of electronics (insulator/metal) for the calculation.')
```

Comment:
Does this input make sense for a molecule?

Reply:
It does not, but I now remember that the default is `METAL`. Any drawback in using metal options for molecules?